api/tests/unit/domain/services/algorithm-methods/pix1d_test.js

```diff
@@ -0,0 +1,180 @@
+          const activity = domainBuilder.buildActivity({ level : Activity.levels.CHALLENGE ,  status : Activity.status.SUCCEEDED});
+          const result = pix1d.getNextActivityLevel([activity]);
+          expect(result).to.equal(undefined);
+        })
+      })
+    })
+    context('when user did the tutorial activity', function() {
+      context('when user succceded the tutorial activity', function() {
+        it('should return the training activity', function() {
+          const activity = domainBuilder.buildActivity({ level : Activity.levels.TUTORIAL , status : Activity.status.SUCCEEDED });
+          const result = pix1d.getNextActivityLevel([activity]);
+          expect(result).to.equal(Activity.levels.TRAINING);
+        })
+      })
+      context('when user succceded the training activity', function() {
+        it('should return the validation activity', function() {
+          const activity1 = domainBuilder.buildActivity({ level : Activity.levels.TUTORIAL });
+          const activity2 = domainBuilder.buildActivity({ level : Activity.levels.TRAINING , status : Activity.status.SUCCEEDED });          
+          const result = pix1d.getNextActivityLevel([activity2, activity1]);
+          expect(result).to.equal(Activity.levels.VALIDATION);
+        })
+      })
+      context('when user succceded the validation activity', function() {
+        it('should end the mission', function() {
+          const activity1 = domainBuilder.buildActivity({ level : Activity.levels.TUTORIAL });
+          const activity2 = domainBuilder.buildActivity({ level : Activity.levels.VALIDATION , status : Activity.status.SUCCEEDED});
+          const result = pix1d.getNextActivityLevel([activity2, activity1]);
+          expect(result).to.equal(undefined);
+        })
+      })
+    })
+  });
+  context('when user fail the activity', function() {
+    context('when user never did the training activity', function() {
+      context('when user failed the validation activity', function() {
+        it('should return the training activity', function() {
+          const activity = domainBuilder.buildActivity({ level : Activity.levels.VALIDATION  , status : Activity.status.FAILED });
+          const result = pix1d.getNextActivityLevel([activity]);
+          expect(result).to.equal(Activity.levels.TRAINING);
+        })
+      })
+      context('when user failed the challenge activity', function() {
+        it('should end the mission', function() {
+          const activity = domainBuilder.buildActivity({ level : Activity.levels.CHALLENGE , status : Activity.status.FAILED });
+          const result = pix1d.getNextActivityLevel([activity]);
+          expect(result).to.equal(undefined);
+        })
+      })
+    })
+    context('when user did the training activity', function() {
+      context('when user fail the training activity', function() {
+        it('should return the tutorial activity', function() {
```

Comment:
Ca ne correspond pas au "when user did the training activity". Soit on en fait 2, soit on remonte celui là en dehors de ce context